NCS.DSS.GDPRUpdateCustomerInformation/Function/GDPRUtility.cs

```diff
@@ -1,26 +1,32 @@
-using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.Functions.Worker;
 using Microsoft.Extensions.Logging;
-using NCS.DSS.GDPRUpdateCustomerInformation.Services;
+using NCS.DSS.GDPRUtility.Services;
 
-namespace NCS.DSS.GDPRUpdateCustomerInformation.Function
+namespace NCS.DSS.GDPRUtility.Function
 {
-    public class GDPRUpdateCustomerInformation
+    public class GDPRUtility
     {
         private readonly IIdentifyAndAnonymiseDataService _identifyAndAnonymiseDataService;
-        private readonly ILogger<GDPRUpdateCustomerInformation> _logger;
+        private readonly ILogger<GDPRUtility> _logger;
 
-        public GDPRUpdateCustomerInformation(IIdentifyAndAnonymiseDataService identifyAndAnonymiseDataService, ILogger<GDPRUpdateCustomerInformation> logger)
+        public GDPRUtility(IIdentifyAndAnonymiseDataService identifyAndAnonymiseDataService, ILogger<GDPRUtility> logger)
         {
             _identifyAndAnonymiseDataService = identifyAndAnonymiseDataService;
             _logger = logger;
         }
 
-        [Function(nameof(GDPRUpdateCustomerInformation))]
-        public async Task<IActionResult> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req)
+        /*
+        In case of failed execution, FixedDelayRetry will retry up to three times after a 15 second delay interval.
+        GDPRUtility runs at 2am every month (on the 1st of the month) as defined in NCRONTAB syntax:
+        {seconds} {minutes} {hours} {day of month} {month} {day-of-week}
+        */
+
+        [Function(nameof(GDPRUtility))]
+        [FixedDelayRetry(3, "00:00:15")] 
```

Comment:
Do we want to retry it on failure?

Reply:
Generally I would say so for something automated of this nature - may reduce the amount of investigations required in future if we can see that a 2nd attempt worked as expected.

Reply:
sure